models/conditional_gans.py

```diff
@@ -65,13 +68,98 @@ class StackGAN1(ConditionalGAN):
     """ Definition for the stage 1 StackGAN """
     def __init__(self, img_size, num_latent_dims, kernel_size,
                  num_filters, reshape_dims):
-        super().__init__(img_size, num_latent_dims, kernel_size,
-                         num_filters, reshape_dims)
+
+        generator = GeneratorStage1(
+            img_size=img_size,
+            num_latent_dims=num_latent_dims,
+            kernel_size=kernel_size,
+            num_filters=num_filters,
+            reshape_dims=reshape_dims
+        )
+
+        discriminator = DiscriminatorStage1(
+            img_size=img_size,
+            kernel_size=kernel_size,
+            num_filters=num_filters
+        )
+
+        super().__init__(
+            generator=generator,
+            discriminator=discriminator,
+            img_size=img_size,
+            num_latent_dims=num_latent_dims,
+            kernel_size=kernel_size,
+            num_filters=num_filters,
+            reshape_dims=reshape_dims
+        )
+
+        self.flatten = Flatten()
+        self.dense_mean = Dense(units=128, activation='relu') # NOTE change units value to variable
+        self.dense_sigma = Dense(units=128, activation='relu') # NOTE change units value to variable
+
+    def generate_conditionals(self, embedding):
+        """ Generate distribution for text embedding.
+            Arguments
+            embedding : Tensor
+                text embedding. Shape (batch_size, feature_size, embedding_size)
+
+            Returns
+            Tensor
+                learnt mean of embedding distribution. Shape (batch_size, reshape_dims/2)
+            Tensor
+                learnt log variance of embedding distribution. Shape (batch_size, reshape_dims/2)
+
+        """
+        x = self.flatten(embedding)
+        mean = self.dense_mean(x)
+        sigma = self.dense_sigma(x)
+        return mean, sigma
+
+    def conditional_augmentation(self, embedding):
+        """ Perform conditional augmentation by sampling embedding.
+            Arguments
+            embedding : Tensor
+                text embedding. Shape (batch_size, feature_size, embedding_size)
+
+            Returns
+            Tensor
+                sampled embedding. Shape (batch_size, reshape_dims/2)
+        """
+        mean, sigma = self.generate_conditionals(embedding)
+        smoothed_embedding = sample_normal(mean, sigma)
+        return smoothed_embedding, mean, sigma
```

Comment:
So do I understand correctly that conditional augmentation is simply sampling a normal with the mean and var of all the elements across all dims in the batch? I'm asking because I can't really see a random variable that is being conditioned on. I will have to re-read the papers, but I thought worth checking because it doesn't seem intuitive to me right now

Reply:
![Screenshot from 2020-04-06 23-21-53](https://user-images.githubusercontent.com/13699844/78606364-7ea11900-785d-11ea-8db9-9e86b3dfa054.png)